keyboards/dz60/keymaps/_bonfire/not-in-use/super-alt-tab.c

```diff
@@ -0,0 +1,37 @@
+/**
+ * Cool Function where a single key does ALT+TAB
+ * From: https://beta.docs.qmk.fm/features/feature_macros#super-alt-tab
+ */
+bool is_alt_tab_active = false;    // ADD this near the begining of keymap.c
+uint16_t alt_tab_timer = 0;        // we will be using them soon.
+
+enum custom_keycodes {            // Make sure have the awesome keycode ready 
+  ALT_TAB = SAFE_RANGE,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {               // This will do most of the grunt work with the keycodes.
+    case ALT_TAB:
+      if (record->event.pressed) {
+        if (!is_alt_tab_active) {
+          is_alt_tab_active = true;
+          register_code(KC_LALT);
+        } 
+        alt_tab_timer = timer_read();
+        register_code(KC_TAB);
+      } else {
+        unregister_code(KC_TAB);
+      }
+      break;
+  }
+  return true;
+}
+
+void matrix_scan_user(void) {     // The very important timer. 
+  if (is_alt_tab_active) {
```

Comment:
you could probably combine the two if clause conditions into one so you don't have to nest it

Reply:
I like that idea - this was test code straight from the docs, and isn't actually implemented in the build of the firmware right now.

but while I have you, maybe you can answer a question for me! Is `matrix_scan_user()` the best place to check a timer's value? Before posting this question here, I also posted in discord, but I am curious...

 The [`matrix_scan_user(void)` docs](https://github.com/qmk/qmk_firmware/blob/master/docs/custom_quantum_functions.md) state,
> This function gets called at every matrix scan, which is basically as often as the MCU can handle. Be careful what you put here, as it will get run a lot.
> You should use this function if you need custom matrix scanning code. It can also be used for custom status output (such as LEDs or a display) or other functionality that you want to trigger regularly even when the user isn't typing.

Right now I have a custom keycode that for a gaming layer (in `keymap-parts/custom-keycodes.c`) that doesn't register escape until it's been held for 450ms. [Example code I repurposed for this task came from the docs](https://beta.docs.qmk.fm/features/feature_macros#advanced-example) and uses `matrix_scan_user(void)` to do a similar task. Is this the best way to do this? I'm unsure what sort of overhead this function adds or how much lifting it's able to do before it affects my board's performance.

Reply:
It's fine to do so. The warning is there more of a "here be dragons" sort of warning.  As long as you're handling the check reasonable, there shouldn't be any issues with using it. 

In fact, there are a number of places where I am doing so, explicitly, for instance.

Reply:
@drashna Ah, thank you! I also saw your reply on discord but failed to respond quickly. I had a question about the code you linked so I will actually comment on it there.

I'm glad to hear the way I am handling this is not going to brick my board or anything! I didn't figure it would, I just always have concerns when it comes to performance warnings. Those are easy things to mess up when you're a green-thumb.

I _am_ still curious if there's a better place to check a running timer than a global function that's called thousands of times per second, whether the timer is relevant or not. For example, is there a way to turn on a specific listener attached to a keycode or a layer change? I know painfully little in this realm and am still trying to read through more of the documentation to solve this myself. My current setup just feels inelegant!